File: windows-lite/src/main/java/com/xushuangbo/clipbridge/windows/util/ServiceAddressFormatter.java
package com.xushuangbo.clipbridge.windows.util;

import java.net.URI;

public final class ServiceAddressFormatter {
    private ServiceAddressFormatter() {
    }

    public static String normalize(String rawAddress) {
        return safeTrim(rawAddress).replaceAll("/+$", "");
    }

    public static String validate(String rawAddress) {
        String normalized = normalize(rawAddress);
        if (normalized.isBlank()) {
            return "请先填写服务地址";
        }
        if (!normalized.startsWith("http://") && !normalized.startsWith("https://")) {
            return "服务地址必须以 http:// 或 https:// 开头";
        }

        URI uri;
        try {
            uri = URI.create(normalized);
        } catch (Exception error) {
            return "服务地址格式不正确";
        }

        if (safeTrim(uri.getHost()).isBlank()) {
            return "服务地址缺少主机名或 IP";
        }
        return "";
    }

    public static String safeTrim(String value) {
        return value == null ? "" : value.trim();
    }
}

File: android/app/src/main/java/com/xushuangbo/clipbridge/core/network/ServiceAddressFormatter.kt
package com.xushuangbo.clipbridge.core.network

import java.net.URI

object ServiceAddressFormatter {
    fun normalize(rawAddress: String): String {
        return rawAddress.trim().removeSuffix("/")
    }

    fun validate(rawAddress: String): String? {
        val normalizedAddress = normalize(rawAddress)

        if (normalizedAddress.isBlank()) {
            return "请先填写服务地址"
        }
        if (!normalizedAddress.startsWith("http://") && !normalizedAddress.startsWith("https://")) {
            return "服务地址必须以 http:// 或 https:// 开头"
        }

        val parsedAddress = runCatching { URI(normalizedAddress) }.getOrNull()
            ?: return "服务地址格式不正确"

        if (parsedAddress.host.isNullOrBlank()) {
            return "服务地址缺少主机名或 IP"
        }

        return null
    }
}
